server/service/src/sync/test/mod.rs

```diff
@@ -142,43 +184,60 @@ pub(crate) async fn check_records_against_database(
                 MasterList(record) => {
-                    assert_eq!(
-                        MasterListRowRepository::new(&con)
-                            .find_one_by_id(&record.id)
-                            .await
-                            .expect(&format!("MasterList row not found: {}", record.id)),
-                        record
-                    )
+                    check_record_by_option_id!(MasterListRowRepository, con, record, "Masterlist")
                 }
+
                 MasterListLine(record) => {
-                    assert_eq!(
-                        MasterListLineRowRepository::new(&con)
-                            .find_one_by_id(&record.id)
-                            .await
-                            .expect(&format!("MasterListLine row not found: {}", record.id)),
-                        record
+                    check_record_by_option_id!(
+                        MasterListLineRowRepository,
+                        con,
+                        record,
+                        "MaseterListLine"
                     )
                 }
-                MasterListNameJoin(record) => assert_eq!(
-                    MasterListNameJoinRepository::new(&con)
-                        .find_one_by_id(&record.id)
-                        .await
-                        .expect(&format!("MasterList row not found: {}", record.id)),
-                    record
+
+                MasterListNameJoin(record) => check_record_by_option_id!(
+                    MasterListNameJoinRepository,
+                    con,
+                    record,
+                    "MasterListNameJoin"
                 ),
+
                 Report(record) => check_record_by_id!(ReportRowRepository, con, record, "Report"),
             }
         }
+
+        for delete in translated_record.deletes {
+            use PullDeleteRecordTable::*;
+            let id = delete.id;
+            match delete.table {
+                Name => {
+                    check_delete_record_by_id!(NameRowRepository, con, id)
+                }
+                Unit => {
+                    check_delete_record_by_id_option!(UnitRowRepository, con, id)
+                }
+                Item => check_delete_record_by_id!(ItemRowRepository, con, id),
+                Store => check_delete_record_by_id!(StoreRowRepository, con, id),
+                MasterList => check_delete_record_by_id_option!(MasterListRowRepository, con, id),
+                MasterListLine => {
+                    check_delete_record_by_id_option!(MasterListLineRowRepository, con, id)
+                }
+                MasterListNameJoin => {
+                    check_delete_record_by_id_option!(MasterListNameJoinRepository, con, id)
+                }
+                Report => check_delete_record_by_id!(ReportRowRepository, con, id),
+                NameStoreJoin => todo!(),
```

Comment:
now there's a handy macro!

Reply:
Yeah, it's very handy, and compiler also does some magic and warnings about this.